packages/core/strapi/tests/filtering.test.e2e.js

```diff
@@ -220,15 +233,26 @@ describe('Filtering API', () => {
         expect(res.body.data).toEqual(expect.arrayContaining(matching));
       });
 
-      test('Should return three matches', async () => {
+      test.each([
+        [{ $notNull: true }],
+        [{ $notNull: 'true' }],
+        [{ $notNull: '1' }],
+        [{ $notNull: 't' }],
+        [{ $notNull: 'anything' }],
+        [{ $notNull: '[]' }],
```

Comment:
Any reason not to test against a realy array not a string ? this is something we can receive (same remark for objects)

Reply:
Mmmh it's a bit tricky here.
- `[]` and `{}` is not stringified by `qs` as it is empty, so it is not present in the query string and cannot be tested here
- However the query can be manually set (by the user) to `filters[code][$null]=`, `qs` will then parse it as `{ filters: { code: { '$null': '' } } }`, not to an empty array of empty object. We already handle the `''` usecase.
- The query can be a non-empty object or array, in this case I can parse it as truthy, will do